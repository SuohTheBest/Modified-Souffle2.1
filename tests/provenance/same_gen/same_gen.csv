John	John
John	Thomas
Cody	Cody
Cody	Lisa
Thomas	John
Thomas	Thomas
Lisa	Cody
Lisa	Lisa
Abraham	Abraham
Abraham	Patricia
Patricia	Abraham
Patricia	Patricia
