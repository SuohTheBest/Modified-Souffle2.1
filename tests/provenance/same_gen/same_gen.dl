// Souffle - A Datalog Compiler
// Copyright (c) 2020, The Souffle Developers. All rights reserved
// Licensed under the Universal Permissive License v 1.0 as shown at:
// - https://opensource.org/licenses/UPL
// - <souffle root>/licenses/SOUFFLE-UPL.txt

// This code tests the provenance explain interface for a simple same generation example,
// including testing the adapted IO system for provenance.

.pragma "provenance" "explain"

.decl parent(x:symbol, y:symbol)
.input parent()

.decl same_gen(x:symbol, y:symbol)
same_gen(x, y) :- parent(x, p), parent(y, p).
same_gen(x, y) :- parent(x, p1), parent(y, p2), same_gen(p1, p2).
.output same_gen()
