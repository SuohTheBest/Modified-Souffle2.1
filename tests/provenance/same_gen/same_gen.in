explain same_gen("John", "Thomas")
