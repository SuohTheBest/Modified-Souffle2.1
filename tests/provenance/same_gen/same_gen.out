                                                parent("Cody", "Melissa") parent("Lisa", "Melissa")  
                                                ------------------------------------------------(R1) 
parent("John", "Cody") parent("Thomas", "Lisa")               same_gen("Cody", "Lisa")               
-------------------------------------------------------------------------------------------------(R2)
                                     same_gen("John", "Thomas")                                      
